cinema/views.py

```diff
@@ -24,28 +27,60 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
+        return [permission() for permission in permission_classes]
 
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
+        return [permission() for permission in permission_classes]
 
 
 class CinemaHallViewSet(viewsets.ModelViewSet):
     queryset = CinemaHall.objects.all()
     serializer_class = CinemaHallSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound()
+        return [permission() for permission in permission_classes]
 
 
 class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.prefetch_related("genres", "actors")
     serializer_class = MovieSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create", "retrieve"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise MethodNotAllowed("PUT", "DELETE")
```

Comment:
In `MovieViewSet`, the `get_permissions` method raises a `MethodNotAllowed` exception for 'PUT' and 'DELETE' actions. Ensure that this aligns with your application's requirements, as it might be more appropriate to handle these actions differently.